### PR TITLE
Add configuration option to create Computer asset from VMs

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

**Comment by trasher** [comment]
I'll take a look to add unit tests

**Comment by trasher** [comment]
Tests added

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 1]
```diff
@@ -3361,4 +3361,65 @@ public function testImportArchive() {
...
+      $nb_vms = countElementsInTable(\ComputerVirtualMachine::getTable());
+      $nb_computers = countElementsInTable(\Computer::getTable());
+      $inventory = new \Glpi\Inventory\Inventory($json);
+
+      if ($inventory->inError()) {
+         foreach ($inventory->getErrors() as $error) {
+            var_dump($error);
+         }
+      }
+      $this->boolean($inventory->inError())->isFalse();
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Inventory.php:3380]:
```suggestion
      $this->boolean($inventory->inError())->isFalse(var_dump($inventory->getErrors()));
```

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 2]
```diff
@@ -3361,4 +3361,65 @@ public function testImportArchive() {
...
+      $this->array($inventory->getErrors())->isEmpty();
+
+      //check inventory metadata
+      $metadata = $inventory->getMetadata();
+      $this->array($metadata)->hasSize(4)
+         ->string['deviceid']->isIdenticalTo('acomputer-2021-01-26-14-32-36')
+         ->string['itemtype']->isIdenticalTo('Computer');
+
+      global $DB;
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Inventory.php:3389]:
```suggestion
```

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 3]
```diff
@@ -3361,4 +3361,65 @@ public function testImportArchive() {
...
+      $this->boolean($conf->saveConf(['vm_as_computer' => 1]))->isTrue();
+
+      $inventory = new \Glpi\Inventory\Inventory($json);
+
+      if ($inventory->inError()) {
+         foreach ($inventory->getErrors() as $error) {
+            var_dump($error);
+         }
+      }
+      $this->boolean($inventory->inError())->isFalse();
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Inventory.php:3409]:
```suggestion
      $this->boolean($inventory->inError())->isFalse(var_dump($inventory->getErrors()));
```

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 4]
```diff
@@ -3361,4 +3361,65 @@ public function testImportArchive() {
...
+      $this->array($inventory->getErrors())->isEmpty();
+
+      //check inventory metadata
+      $metadata = $inventory->getMetadata();
+      $this->array($metadata)->hasSize(4)
+         ->string['deviceid']->isIdenticalTo('acomputer-2021-01-26-14-32-36')
+         ->string['itemtype']->isIdenticalTo('Computer');
+
+      global $DB;
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Inventory.php:3418]:
```suggestion
```

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 5]
```diff
@@ -3361,4 +3361,65 @@ public function testImportArchive() {
...
+      $this->integer($count_vms)->isIdenticalTo(6);
+
+      $nb_vms = countElementsInTable(\ComputerVirtualMachine::getTable());
+      $nb_computers = countElementsInTable(\Computer::getTable());
+      $inventory = new \Glpi\Inventory\Inventory($json);
+
+      if ($inventory->inError()) {
+         foreach ($inventory->getErrors() as $error) {
+            var_dump($error);
+         }
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Inventory.php:3378]:
```suggestion
         $this->dump($inventory->getErrors());
```

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 6]
```diff
@@ -3361,4 +3361,65 @@ public function testImportArchive() {
...
+      $this->login();
+      $conf = new \Glpi\Inventory\Conf();
+      $this->boolean($conf->saveConf(['vm_as_computer' => 1]))->isTrue();
+
+      $inventory = new \Glpi\Inventory\Inventory($json);
+
+      if ($inventory->inError()) {
+         foreach ($inventory->getErrors() as $error) {
+            var_dump($error);
+         }
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Inventory.php:3407]:
```suggestion
         $this->dump($inventory->getErrors());
```

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**